File: app/src/main/java/com/creat/motiv/Adapters/RecyclerReferencesAdapter.java
package com.creat.motiv.Adapters;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.creat.motiv.Beans.Artists;
import com.creat.motiv.R;
import com.creat.motiv.Utils.Tools;

import java.util.ArrayList;


public class RecyclerReferencesAdapter extends RecyclerView.Adapter<RecyclerReferencesAdapter.MyViewHolder>  {
    private ArrayList<Artists> mData;
    private Activity mActivity;


    public RecyclerReferencesAdapter(Activity mActivity) {
        this.mActivity = mActivity;
        mData = Tools.references(mActivity);
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view;
        LayoutInflater mInflater = LayoutInflater.from(mActivity);
        view = mInflater.inflate(R.layout.references_layout, parent, false);

        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final MyViewHolder holder, final int position) {
        final Artists a = mData.get(holder.getAdapterPosition());
        holder.nome.setText(a.getNome());
        holder.nome.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Uri uri = Uri.parse(a.getUri());
                Intent intent = new Intent(Intent.ACTION_VIEW, uri);
                mActivity.startActivity(intent);
            }
        });
        Log.println(Log.INFO,"REFERENCE color", String.valueOf(a.getColor()));
        holder.nome.setTextColor(a.getColor());



    }



    @Override
    public int getItemCount() {

        return mData.size();
    }


    class MyViewHolder extends RecyclerView.ViewHolder {
        TextView nome;
        MyViewHolder(View view) {
            super(view);
            nome = view.findViewById(R.id.reference);



        }
    }
}

File: app/src/main/java/com/creat/motiv/view/adapters/RecyclerReferencesAdapter.kt
package com.creat.motiv.view.adapters

import android.content.Intent
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.creat.motiv.R
import com.creat.motiv.databinding.ReferencesLayoutBinding
import com.creat.motiv.features.about.data.Reference
import com.ilustris.motiv.base.utils.WEB_URL
import com.ilustris.animations.slideInBottom


class RecyclerReferencesAdapter(private val references: List<Reference>) :
    RecyclerView.Adapter<RecyclerReferencesAdapter.MyViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        return MyViewHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.references_layout, parent, false)
        )
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.bind()
    }

    override fun getItemCount(): Int {
        return references.size
    }


    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind() {
            ReferencesLayoutBinding.bind(itemView).run {
                val context = itemView.context
                val reference = references[bindingAdapterPosition]
                referenceBackground.setBackgroundResource(reference.background)
                referenceCard.setOnClickListener {
                    val uri = Uri.parse("$WEB_URL${reference.url}")
                    val intent = Intent(Intent.ACTION_VIEW, uri)
                    context.startActivity(intent)
                }
                referenceCard.slideInBottom()
            }
        }
    }
}
